src/hydra/dataProvider.ts

```diff
@@ -268,22 +299,33 @@ function dataProvider(
 
   const transformReactAdminDataToRequestBody = (
     resource: string,
-    data: object,
-    extraInformation: { hasFileField?: any },
-  ): Promise<any> => {
-    /** @type {Resource} */
+    data: { [key: string]: unknown } | unknown,
+    extraInformation: { hasFileField?: boolean },
+  ): Promise<XMLHttpRequestBodyInit> => {
     const apiResource = apiSchema.resources.find(
       ({ name }) => resource === name,
     );
     if (undefined === apiResource) {
-      return Promise.resolve(data);
+      return Promise.resolve(data as FormData);
     }
 
-    return convertReactAdminDataToHydraData(apiResource, data).then((data) => {
+    return convertReactAdminDataToHydraData(
+      apiResource,
+      data as { [key: string]: unknown },
+    ).then((data) => {
       const values = Object.values(data);
-      const containFile = (element) =>
+      const containFile = (element: unknown) =>
         isPlainObject(element) &&
-        Object.values(element).some((value) => value instanceof File);
+        Object.values(element as { [key: string]: unknown }).some(
+          (value) => value instanceof File,
+        );
+      type toJSONObject = { toJSON(): string };
```

Comment:
move to types?

Reply:
This is really an internal type, I think it can stay here.